packages/tanstack-react-query/src/internals/createOptionsProxy.ts

```diff
@@ -98,14 +100,20 @@ export interface DecorateQueryProcedure<TDef extends ResolverDef> {
    *
    * @see https://tanstack.com/query/latest/docs/framework/react/guides/query-keys
    */
-  queryKey: (input?: TDef['input']) => TRPCQueryKey;
+  queryKey: (
+    input?: TDef['input'],
+    // eslint-disable-next-line @typescript-eslint/no-redundant-type-constituents
```

Comment:
I'm not sure why this is complaining, DataTag has a default of `unknown` and errorShape has a default of `any` - the types themselves seem to work but ts-eslint complains about this